subworkflows/icgc-argo-workflows/stage_input/main.nf

```diff
@@ -62,69 +106,34 @@ workflow STAGE_INPUT {
           sample:row.sample,
           sex:row.sex,
           status:row.status.toInteger(), 
-          qc_tools:row.qc_tools, 
-          data_type:'tgz'], file(row.qc_file))
+          qc_tools:row.qc_tools,
+          genome_build:row.genome_build,
+          experiment:row.experiment,
+          data_type:'tgz'],
+          [file(row.qc_file)],
+          row.analysis_json
+          )
       }
     }
     .set { ch_input_sample }
 
-    PREP_SAMPLE.out.sample_sheet_csv
-    .collectFile(keepHeader: true)
-    .splitCsv(header:true)
-    .map{ row ->
-      if (row.analysis_type == "sequencing_experiment") {
-        tuple([
-          id:"${row.sample}-${row.lane}".toString(), 
-          study_id:row.study_id,
-          patient:row.patient,
-          sex:row.sex,
-          status:row.status.toInteger(),
-          sample:row.sample, 
-          read_group:row.read_group.toString(), 
-          data_type:'json', 
-          size:1, 
-          numLanes:row.read_group_count], 
-          file(row.analysis_json)) 
-      }
-      else if (row.analysis_type == "sequencing_alignment") {
-        tuple([
-          id:"${row.sample}".toString(),
-          study_id:row.study_id,
-          patient:row.patient,
-          sample:row.sample,
-          sex:row.sex,
-          status:row.status.toInteger(), 
-          data_type:'json'], 
-          file(row.analysis_json))
-      }
-      else if (row.analysis_type == "variant_calling") {
-        tuple([
-          id:"${row.sample}".toString(),
-          study_id:row.study_id, 
-          patient:row.patient,
-          sample:row.sample, 
-          variantcaller:row.variantcaller, 
-          data_type:'json'], file(row.analysis_json))
-      }
-      else if (row.analysis_type == "qc_metrics") {
-        tuple([
-          id:"${row.sample}".toString(),
-          study_id:row.study_id, 
-          patient:row.patient,
-          sample:row.sample,
-          sex:row.sex,
-          status:row.status.toInteger(),  
-          qc_tools:row.qc_tools, 
-          data_type:'json'], file(row.analysis_json))
+    ch_input_sample.map{ meta,files,analysis ->
+      if (analysis){
+        tuple([meta,file(analysis)])
+      } else {
+        tuple([meta,null])
       }
     }
-    .set { ch_meta_analysis }
+    .set{ ch_meta_analysis }
+
+    ch_input_sample.map{ meta,files,analysis -> tuple([meta,files])}.set{ch_files}
+
+    //ch_meta_analysis.subscribe { println "$it" }
+    //ch_files.subscribe { println "$it" }
 
     emit:
-    analysis_json = SONG_SCORE_DOWNLOAD.out.analysis_json  // channel: [ analysis_json ] 
     meta_analysis = ch_meta_analysis         // channel: [ val(meta), analysis_json]
-    sample_files  = ch_input_sample          // channel: [ val(meta), [ files ] ]
-    input_files = SONG_SCORE_DOWNLOAD.out.files // channel: [files]
+    sample_files  = ch_files          // channel: [ val(meta), [ files ] ]
```

Comment:
most down streaming modules expect the input files to be flat for example: tuple [val(meta), bam, bai] instead of tuple[val(meta), [bam, bai]], any thoughts of how to support it?